### PR TITLE
Updated template to use routing lib 'comidi' instead of 'compojure'

### PR DESCRIPTION
**Pull request by HairyMike**

### CODE REVIEW

File: resources/leiningen/new/trapperkeeper/src/web_service.clj [hunk 1]
```diff
@@ -14,8 +15,11 @@
     (let [url-prefix (get-route this)]
       (add-ring-handler
         this
-        (compojure/context url-prefix []
-          (core/app (tk-services/get-service this :HelloService))))
+        (->
+          (comidi/routes->handler (comidi/context url-prefix
+            (core/app (tk-services/get-service this :HelloService))))
+          ; Ring middleware
+          (wrap-params)))
```
Comment (scottyw) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: Should ```wrap-params``` be called i.e. ```(wrap-params)``` vs ```wrap-params```?

File: resources/leiningen/new/trapperkeeper/src/web_service.clj [hunk 2]
```diff
@@ -14,8 +15,11 @@
     (let [url-prefix (get-route this)]
       (add-ring-handler
         this
-        (compojure/context url-prefix []
-          (core/app (tk-services/get-service this :HelloService))))
+        (->
```
Comment (scottyw) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: Might be nice to thread ```core/app``` through ```context```, ```route->handler``` and ```wrap-params```?

**Comment by scottyw** [comment]
Very nice

File: resources/leiningen/new/trapperkeeper/src/web_service.clj [hunk 3]
```diff
@@ -14,8 +15,10 @@
     (let [url-prefix (get-route this)]
       (add-ring-handler
         this
-        (compojure/context url-prefix []
-          (core/app (tk-services/get-service this :HelloService))))
+        (->> (core/app (tk-services/get-service this :HelloService))
+             (comidi/context url-prefix)
+             (comidi/routes->handler)
+             (wrap-params)))
```
Comment (dmcpl) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: `wrap-params` was not in the previous template. It is also unnecessary as we have no parameters in the example services. Should we lose this?

**Comment by dmcpl** [comment]
We should bring the Trapperkeeper versions up to the latest but this could be a separate PR.

File: resources/leiningen/new/trapperkeeper/project.clj [hunk 4]
```diff
@@ -15,11 +15,12 @@
                  ;; explicit versions of deps that would cause transitive dep conflicts
                  [org.clojure/tools.reader "1.0.0-beta1"]
                  [slingshot "0.12.2"]
-                 [clj-time "0.9.0"]
+                 [clj-time "0.10.0"]
+                 [puppetlabs/kitchensink "1.3.0"]
```
Comment (dmcpl) [inline resources/leiningen/new/trapperkeeper/project.clj]: Is this used?

Comment (HairyMike) [inline resources/leiningen/new/trapperkeeper/project.clj]: there is a transitive dep conflict without it

File: resources/leiningen/new/trapperkeeper/src/web_service.clj (hunk 3 above)
Comment (HairyMike) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: done

File: resources/leiningen/new/trapperkeeper/src/web_service.clj (hunk 2 above)
Comment (HairyMike) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: done

File: resources/leiningen/new/trapperkeeper/src/web_service.clj (hunk 1 above)
Comment (HairyMike) [inline resources/leiningen/new/trapperkeeper/src/web_service.clj]: removed wrap-params altogether as per DMC's comment

**Comment by scottyw** [comment]
👍

**Pull request merged**